spread/project_test.go

```diff
@@ -78,19 +77,42 @@ suites:
  tests/:
   summary: mock tests
 `)
-	tmpdir := c.MkDir()
-	err := ioutil.WriteFile(filepath.Join(tmpdir, "spread.yaml"), spreadYaml, 0644)
-	c.Assert(err, IsNil)
-	err = os.MkdirAll(filepath.Join(tmpdir, "tests"), 0755)
-	c.Assert(err, IsNil)
 
-	prj, err := spread.Load(tmpdir)
-	c.Assert(err, IsNil)
-	backend := prj.Backends["google"]
-	c.Check(backend.Name, Equals, "google")
-	c.Check(backend.Systems["system-1"].Plan, Equals, "global-plan")
-	c.Check(backend.Systems["system-2"].Plan, Equals, "plan-for-2")
-	c.Check(backend.Systems["system-3"].Plan, Equals, "global-plan")
+	for _, tc := range []struct {
+		name        string // the file name to write the project to
+		prjFilename string // the file name specified by the user
+		errMsg      string // the expected error message
+	}{
+		{"spread.yaml", "", ""},
+		{".spread.yaml", "", ""},
+		{"other.yaml", "", "cannot load project from .*: cannot find spread.yaml or .spread.yaml"},
+		{"custom.yaml", "custom.yaml", ""},
```

Comment:
While it doesn't really affect this implementation, it might be a good idea to have a test that makes the file in a subdirectory so that if the implementation of `Load()` changes we don't accidentally drop that use case.

Reply:
This was a necessary test because loading from a subdir was actually broken. When no custom filename is specified, the project path is where `spread.yaml` is located, and can be in a parent directory. But if we specify a filename, potentially with a relative or absolute path, we'll keep the project path in the current directory without verifying parents.